apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -147,6 +162,33 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     }
 
     if (!packageDescription.dependencies.hasOwnProperty(dependencyName)) {
+      if (versionRange) {
+        // this means the current temp project doesn't provide this dependency,
+        // however, we may be able to use a different version. we prefer the latest version
+        const minimumVersion: string = '0.0.0';
```

Comment:
>'0.0.0' [](start = 39, length = 7)

This value will not be comparable with other forms such as pre-release versions.  It would be safer to start with "undefined" and then explicitly assign the first encountered value (skipping semver.gt() in that case).